core/src/main/scala/org/apache/spark/scheduler/SparkListener.scala

```diff
@@ -431,23 +481,23 @@ abstract class SparkListener extends SparkListenerInterface {
 
   override def onExecutorRemoved(executorRemoved: SparkListenerExecutorRemoved): Unit = { }
 
-  override def onExecutorBlacklisted(
-      executorBlacklisted: SparkListenerExecutorBlacklisted): Unit = { }
+  override def onExecutorExcluded(
+      executorExcluded: SparkListenerExecutorExcluded): Unit = { }
 
-  def onExecutorBlacklistedForStage(
-      executorBlacklistedForStage: SparkListenerExecutorBlacklistedForStage): Unit = { }
+  override def onExecutorExcludedForStage(
+      executorExcludedForStage: SparkListenerExecutorExcludedForStage): Unit = { }
 
-  def onNodeBlacklistedForStage(
-      nodeBlacklistedForStage: SparkListenerNodeBlacklistedForStage): Unit = { }
+  override def onNodeExcludedForStage(
+      nodeExcludedForStage: SparkListenerNodeExcludedForStage): Unit = { }
 
-  override def onExecutorUnblacklisted(
-      executorUnblacklisted: SparkListenerExecutorUnblacklisted): Unit = { }
+  override def onExecutorUnexcluded(
+      executorUnexcluded: SparkListenerExecutorUnexcluded): Unit = { }
 
-  override def onNodeBlacklisted(
-      nodeBlacklisted: SparkListenerNodeBlacklisted): Unit = { }
+  override def onNodeExcluded(
+      nodeExcluded: SparkListenerNodeExcluded): Unit = { }
 
-  override def onNodeUnblacklisted(
-      nodeUnblacklisted: SparkListenerNodeUnblacklisted): Unit = { }
+  override def onNodeUnexcluded(
+      nodeUnexcluded: SparkListenerNodeUnexcluded): Unit = { }
 
```

Comment:
I am catching up on this PR; this would be an incompatible source change right ?
Existing listeners which override `SparkListener`, particularly `override def onNodeBlacklisted` (for ex) will now fail compilation and will need to be updated to `override def onNodeExcluded` (and so on) - in a minor release.

Given we are not removing the event, do we want to keep both - but mark `*blaclisted` api's as deprecated and to be removed in next major release ?

Or are we doing this to force users to change ? In particular, to ensure that expectation of invocation of `*blaclisted` api's will not happen ?

Reply:
Also catching up, but, yes please merely deprecate things not remove them as part of this, where they could possibly be called or extended by apps

Reply:
Updated to put back